app/models/cda/prosecution_case.rb

```diff
@@ -1,5 +1,15 @@
 module Cda
   class ProsecutionCase < BaseModel
     has_many :defendant_summaries, class_name: 'Cda::Defendant'
+
+    def self.find(urn)
+      response = Cda::ProsecutionCaseSearchResponse.find(
```

Comment:
why does it call the `Cda::ProsecutionCaseSearchResponse` here?

Can we use the find associated the the currect`ProsecutionCase`?
something like this: 
```
def self.find_by_urn(urn)
  find(:one,
        params: { filter: { prosecution_case_reference: urn } },
        from: '/api/internal/v2/prosecution_cases'
      )
 end
 ```
 🤔

Reply:
You're right, this logic is in the wrong place. The problem is that the `prosecution_cases` endpoint doesn't return an array of prosecution cases, it returns a data structure with attributes like `total_results` etc. We need to unwrap that. But that should happen in a service. I'll move it.